x/protocolpool/keeper/keeper.go

```diff
@@ -306,9 +315,9 @@ func (k Keeper) iterateAndUpdateFundsDistribution(ctx context.Context, toDistrib
 	return k.ToDistribute.Set(ctx, math.ZeroInt())
 }
 
-func (k Keeper) claimFunds(ctx context.Context, recipient sdk.AccAddress) (amount sdk.Coin, err error) {
+func (k Keeper) claimFunds(ctx context.Context, recipient sdk.AccAddress, recipientAddr string) (amount sdk.Coin, err error) {
```

Comment:
we could just keep one recipient arg (possibly string) and use StringToBytes check here instead in msg_server code, having string and bytes are redundant.

Reply:
nice catch!